include/alpaka/dev/DevGenericSycl.hpp

```diff
@@ -0,0 +1,165 @@
+/* Copyright 2021 Jan Stephan
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#ifdef ALPAKA_ACC_SYCL_ENABLED
+
+#    include <alpaka/acc/Traits.hpp>
+#    include <alpaka/core/Common.hpp>
+#    include <alpaka/core/Sycl.hpp>
+#    include <alpaka/dev/Traits.hpp>
+#    include <alpaka/mem/buf/Traits.hpp>
+#    include <alpaka/pltf/Traits.hpp>
+#    include <alpaka/queue/Properties.hpp>
+#    include <alpaka/queue/Traits.hpp>
+#    include <alpaka/wait/Traits.hpp>
+
+#    include <CL/sycl.hpp>
+
+#    include <shared_mutex>
+#    include <vector>
+
+namespace alpaka
+{
+    namespace experimental
+    {
+        template<typename TElem, typename TDim, typename TIdx, typename TDev>
+        class BufGenericSycl;
+
+        template<typename TDev>
+        class QueueGenericSyclBlocking;
+
+        template<typename TDev>
+        class QueueGenericSyclNonBlocking;
+
+        //! The SYCL device handle.
+        template<typename TPltf>
+        class DevGenericSycl
+            : public concepts::Implements<ConceptCurrentThreadWaitFor, DevGenericSycl<TPltf>>
+            , public concepts::Implements<ConceptDev, DevGenericSycl<TPltf>>
+        {
+        public:
+            DevGenericSycl(sycl::device device, sycl::context context) : m_device{device}, m_context{context}
+            {
+            }
+
+            auto operator==(DevGenericSycl const& rhs) const
+            {
+                return (rhs.m_device == m_device);
+            }
```

Comment:
Prefer a hidden friend for comparison operators.

```suggestion
            friend auto operator==(DevGenericSycl const& lhs, DevGenericSycl const& rhs)
            {
                return lhs.m_device == rhs.m_device;
            }
```